src/ptvsd/adapter/server.py

```diff
@@ -0,0 +1,144 @@
+            "supportsReadMemoryRequest": False,
+            "supportsDisassembleRequest": False,
+            "exceptionBreakpointFilters": [],
+            "additionalModuleColumns": [],
+            "supportedChecksumAlgorithms": [],
+        }
+
+    def __init__(self, session, stream):
+        super(Server, self).__init__(session, stream)
+
+        self.pid = None
+        """Process ID of the debuggee process, as reported by the server."""
+
+        assert not session.server
+        session.server = self
+
+    def initialize(self, request):
+        assert request.is_request("initialize")
+        request = self.channel.propagate(request)
+        request.wait_for_response()
+        self.capabilities = self.Capabilities(self, request.response)
+
+    # Generic request handler, used if there's no specific handler below.
+    @message_handler
+    def request(self, request):
+        # Do not delegate requests from the server by default. There is a security
+        # boundary between the server and the adapter, and we cannot trust arbitrary
+        # requests sent over that boundary, since they may contain arbitrary code
+        # that the IDE will execute - e.g. "runInTerminal". The adapter must only
+        # propagate requests that it knows are safe.
+        raise request.isnt_valid(
+            "Requests from the debug server to the IDE are not allowed."
+        )
+
+    # Generic event handler, used if there's no specific handler below.
+    @message_handler
+    def event(self, event):
+        self.ide.propagate_after_start(event)
+
+    @message_handler
+    def initialized_event(self, event):
+        # pydevd doesn't send it, but the adapter will send its own in any case.
+        pass
+
+    @message_handler
+    def process_event(self, event):
+        self.pid = event("systemProcessId", int)
+
+        if self.launcher:
+            assert self.session.pid is not None
+        else:
+            assert self.session.pid is None
+        if self.session.pid is not None and self.session.pid != self.pid:
+            event.cant_handle(
+                '"process" event mismatch: launcher reported "systemProcessId":{0}, '
+                'but server reported "systemProcessId":{1}',
+                self.session.pid,
+                self.pid,
+            )
+        else:
+            self.session.pid = self.pid
+
+        # If there is a launcher, it's handling the process event.
+        if not self.launcher:
+            self.ide.propagate_after_start(event)
+
+    @message_handler
+    def continued_event(self, event):
+        # https://github.com/microsoft/ptvsd/issues/1530
+        #
+        # DAP specification says that a step request implies that only the thread on
+        # which that step occurred is resumed for the duration of the step. However,
+        # for VS compatibility, pydevd can operate in a mode that resumes all threads
+        # instead, and it is the default mode for all clients.
+        #
```

Comment:
```suggestion
        #
        # This can be changed using "steppingResumesAllThreads" in launch.json.
        # Default is true. To resume the stepped thread only set it to false. 
        #
```